hnn_core/dipole.py

```diff
@@ -101,12 +101,10 @@ def average_dipoles(dpls):
         A new dipole object with each component of `dpl.data` representing the
         average over the same components in the input list
     """
-    # need at least one Dipole to get times
-    if len(dpls) < 2:
-        raise ValueError("Need at least two dipole object to compute an"
-                         " average")
-
     for dpl_idx, dpl in enumerate(dpls):
+        if not isinstance(dpl, Dipole):
+            raise ValueError("Need at least one dipole object to return a"
+                             " dipole")
```

Comment:
there needs to be a test to check that this works. Or is there already a test? Also, I think the text should be changed to reflect the fact that all objects in the list must instances of `Dipole`. Also, when you raise an error, you should tell the user what you got. Something like:

```suggestion
            raise ValueError(f"All elements in the list should be instances of Dipole. Got {type(dpl)}")
```

Reply:
Yea, this test already exists. It's under line 51, the comment `# average an n_of_1 dipole list` which calls `average_dipoles()` on a list with only one dipole.